ecs-agent/netlib/platform/common_linux.go

```diff
@@ -0,0 +1,200 @@
+	// the corresponding network interface.
+	// Empty Name fields in network interface names indicate that all
+	// interfaces share the same network namespace. This use case is
+	// utilized by certain internal teams like EKS on Fargate.
+	if len(taskPayload.ElasticNetworkInterfaces) == 1 ||
+		aws.StringValue(taskPayload.ElasticNetworkInterfaces[0].Name) == "" {
+		primaryNetNS, err := c.buildSingleNSNetConfig(taskID,
+			0,
+			taskPayload.ElasticNetworkInterfaces,
+			taskPayload.ProxyConfiguration)
+		if err != nil {
+			return nil, err
+		}
+
+		return []*tasknetworkconfig.NetworkNamespace{primaryNetNS}, nil
+	}
+
+	// Create a map for easier lookup of ENIs by their names.
+	ifNameMap := make(map[string]*ecsacs.ElasticNetworkInterface, len(taskPayload.ElasticNetworkInterfaces))
+	for _, iface := range taskPayload.ElasticNetworkInterfaces {
+		ifNameMap[networkinterface.GetInterfaceName(iface)] = iface
+	}
+
+	// Proxy configuration is not supported yet in a multi-ENI / multi-NetNS task.
+	if taskPayload.ProxyConfiguration != nil {
+		return nil, errors.New("unexpected proxy config found")
+	}
+
+	// The number of network namespaces required to create depends on the
+	// number of unique interface names list across all container definitions
+	// in the task payload. Meaning if two containers are linked with the same
+	// set of network interface names, both those containers share the same namespace.
+	// If not, they reside in two different namespaces. Also, an interface can only
+	// belong to one NetworkNamespace object.
+
+	var netNSs []*tasknetworkconfig.NetworkNamespace
+	nsIndex := 0
+	// Loop through each container definition and their network interfaces.
+	for _, container := range taskPayload.Containers {
+		// ifaces holds all interfaces associated with a particular container.
+		var ifaces []*ecsacs.ElasticNetworkInterface
+		for _, ifNameP := range container.NetworkInterfaceNames {
+			ifName := aws.StringValue(ifNameP)
+			if iface := ifNameMap[ifName]; iface != nil {
+				ifaces = append(ifaces, iface)
+				// Remove ENI from map to indicate that the ENI is assigned to
+				// a namespace.
+				delete(ifNameMap, ifName)
+			} else {
+				// If the ENI does not exist in the lookup map, it means the ENI
+				// is already assigned to a namespace. The container will be run
+				// in the same namespace.
+				break
+			}
+		}
+
+		if len(ifaces) == 0 {
+			continue
+		}
+
+		netNS, err := c.buildSingleNSNetConfig(taskID, nsIndex, ifaces, nil)
+		if err != nil {
+			return nil, err
+		}
+		netNSs = append(netNSs, netNS)
+		nsIndex += 1
+	}
+
+	return netNSs, nil
+}
+
+func (c *common) buildSingleNSNetConfig(
+	taskID string,
+	index int,
+	networkInterfaces []*ecsacs.ElasticNetworkInterface,
+	proxyConfig *ecsacs.ProxyConfiguration) (*tasknetworkconfig.NetworkNamespace, error) {
+	var primaryIF *networkinterface.NetworkInterface
+	var ifaces []*networkinterface.NetworkInterface
+	lowestIdx := int64(10)
```

Comment:
Magic number - can we add comment or move it to const?

Reply:
Thanks for catching, will update.